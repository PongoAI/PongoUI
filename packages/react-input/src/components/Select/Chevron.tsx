import * as React from 'react';
import { makeStyles } from '@griffel/react';
import { tokens } from '@pongo-ui/react-theme';

const useStyles = makeStyles({
  disabled: {
    stroke: tokens.inheritDisabled,
  },
  enabled: {
    stroke: tokens.inherit,
  },
});

type ChevronState = {
  /**
   * Whether content is visible.
   */
  visible: boolean;

  /**
   * Whether the Chevron component is disabled.
   */
  disabled?: boolean;
};

export const Chevron = (props: ChevronState) => {
  const { visible, disabled } = props;

  const styles = useStyles();

  const leftLineStyles = {
    transformOrigin: 'center',
    transform: visible ? 'rotate(-45deg)' : 'rotate(45deg)',
    transition: 'transform .1s linear',
  };

  const rightLineStyles = {
    transformOrigin: 'center',
    transform: visible ? 'rotate(45deg)' : 'rotate(-45deg)',
    transition: 'transform .1s linear',
  };

  const wrapperStyles = {
    transform: visible ? 'translateY(-20%)' : 'translateY(20%)',
    transition: 'transform .1s linear',
  };

  return (
    <svg
      width="20px"
      height="20px"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
      className={disabled ? styles.disabled : styles.enabled}
    >
      <g style={wrapperStyles}>
        <line x1="1" y1="10" x2="10" y2="10" strokeWidth="2px" style={leftLineStyles} strokeLinecap="round" />
        <line x1="10" y1="10" x2="19" y2="10" style={rightLineStyles} strokeWidth="2px" strokeLinecap="round" />
      </g>
    </svg>
  );
};
