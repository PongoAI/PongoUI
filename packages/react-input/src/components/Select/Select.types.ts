import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
import type { InputWrapperCommons } from '../InputWrapper';
import { InputWrapper } from '../InputWrapper';
import { MenuPopover, MenuList } from '@pongo-ui/react-menu';

export type SelectSlots = {
  /**
   * The root of the Input. This is the wrapper component that handles it's appearance.
   */
  root: NonNullable<Slot<typeof InputWrapper>>;

  /**
   * Select that displays the active value of the Select component and handles accessability.
   */
  select: NonNullable<Slot<'select'>>;

  /**
   * The wrapper component that manages state for a popup MenuList and MenuTrigger
   */
  menu: any;

  /**
   * Wrapper for the Menu trigger element.
   */
  menuTrigger: any;

  /**
   *  Wrapper for `MenuList` used to add styling and interaction support.
   */
  menuPopover: NonNullable<Slot<typeof MenuPopover>>;

  /**
   * Wrapper for `MenuItems`. This is what handles the items context.
   */
  menuList: NonNullable<Slot<typeof MenuList>>;
};

type SelectCommons = {
  /**
   * The starting value for a uncontrolled Input. If `true` then the Input will be enabled.
   * Mutually exclusive with `value` prop.
   *
   * @defaultValue ''
   */
  defaultValue?: string[];

  /**
   * The current value for a controlled Input. If `true` then the Input will be enabled.
   * Mutually exclusive with `defaultValue` prop.
   */
  value?: string[];

  /**
   * Callback to be called when the value changes.
   */
  onCheckedValueChange?: (ev: React.MouseEvent | React.KeyboardEvent, data: { checkedItems: string[] }) => void;

  /**
   * The content to render before the Select.
   */
  contentBefore?: string | number | JSX.Element;

  /**
   * The content to render after the Select.
   */
  contentAfter?: string | number | JSX.Element;

  /**
   * The helper text to render below the Input.
   */
  helperText?: string;

  /**
   * The label for the Input.
   */
  label?: string;

  /**
   * The placeholder text for the Input.
   */
  placeholder?: string;
};

export type SelectProps = ComponentProps<Partial<SelectSlots>> & SelectCommons & InputWrapperCommons;

export type SelectState = ComponentState<SelectSlots> & SelectCommons & InputWrapperCommons;
