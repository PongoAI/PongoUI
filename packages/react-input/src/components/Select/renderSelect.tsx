import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import { Menu, MenuTrigger, MenuList, MenuPopover, MenuItem, MenuItemRadio } from '@pongo-ui/react-menu';
import type { SelectSlots, SelectState } from './Select.types';

export const renderSelect = (state: SelectState) => {
  const { slots, slotProps } = getSlots<SelectSlots>(state);

  return (
    <slots.menu {...slotProps.menu}>
      <slots.menuTrigger {...slotProps.menuTrigger}>
        <slots.root {...slotProps.root}>
          <slots.input {...slotProps.input} />
        </slots.root>
      </slots.menuTrigger>
      <slots.menuPopover {...slotProps.menuPopover}>
        <slots.menuList {...slotProps.menuList}>
          <MenuItemRadio name="font" value="segoe">
            Segoe
          </MenuItemRadio>
          <MenuItemRadio name="font" value="calibri">
            Calibri
          </MenuItemRadio>
          <MenuItemRadio name="font" value="arial">
            Arial
          </MenuItemRadio>
        </slots.menuList>
      </slots.menuPopover>
    </slots.menu>
  );
};
