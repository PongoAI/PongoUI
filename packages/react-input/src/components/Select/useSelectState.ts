import type { SelectState } from './Select.types';

export const useSelectState = (state: SelectState) => {
  return state;
};
