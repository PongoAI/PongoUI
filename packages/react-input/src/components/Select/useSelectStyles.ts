import { shorthands, makeStyles, mergeClasses } from '@griffel/react';
import type { SelectState } from './Select.types';
import { tokens } from '@cebus/react-theme';
import { inputLineHeight } from '../InputWrapper';

const useSelectElementStyles = makeStyles({
  select: {
    ...shorthands.margin('0px'),
    ...shorthands.padding('0px', '10px'),
    ...shorthands.borderStyle('none'),
    ...shorthands.borderRadius(tokens.rounded),
    display: 'flex',
    boxSizing: 'border-box',
    flexGrow: 1,
    minHeight: `var(${inputLineHeight})`,
    minWidth: '100px',
    height: '100%',
    alignItems: 'center',
    backgroundColor: 'transparent',
    ':focus-visible': {
      outlineStyle: 'none',
    },
    '-webkit-appearance': 'none',
    '-moz-appearance': 'none',
    '-o-appearance': 'none',
  },

  small: {
    fontSize: tokens.fontSize200,
  },

  medium: {
    fontSize: tokens.fontSize300,
  },

  large: {
    fontSize: tokens.fontSize400,
  },

  danger: {
    color: tokens.textColor,
    '::placeholder': {
      color: tokens.danger,
    },
  },

  enabled: {
    cursor: 'pointer',
    color: tokens.textColor,
    '::placeholder': {
      color: tokens.inherit,
      opacity: 1,
    },
  },

  disabled: {
    cursor: 'not-allowed',
    color: tokens.inheritDisabled,
    '::placeholder': {
      color: tokens.inheritDisabled,
      opacity: 1,
    },
  },

  dangerDisabled: {
    cursor: 'not-allowed',
    color: tokens.dangerDisabled,
    '::placeholder': {
      color: tokens.dangerDisabled,
      opacity: 1,
    },
  },

  labelLowerText: {
    paddingTop: '10px',
  },

  labelPlaceholderFocus: {
    '::placeholder': {
      opacity: '0',
    },
    ':focus': {
      '::placeholder': {
        opacity: '1',
        color: tokens.inherit,
      },
    },
  },
});

export const useSelectStyles = (state: SelectState) => {
  const rootStyles = useSelectElementStyles();

  state.selectValue.className = mergeClasses(
    rootStyles.select,
    state.disabled ? rootStyles.disabled : rootStyles.enabled,
    !state.disabled && state.danger && rootStyles.danger,
    state.disabled && state.danger && rootStyles.dangerDisabled,
    state.label && state.placeholder && !state.root.contentBefore && rootStyles.labelPlaceholderFocus,
    state.label && (state.appearance === 'filled' || state.appearance === 'standard') && rootStyles.labelLowerText,
    rootStyles[state.size!],
    state.selectValue.className,
  );

  state.root.className = mergeClasses(state.disabled ? rootStyles.disabled : rootStyles.enabled, state.root.className);

  return state;
};
