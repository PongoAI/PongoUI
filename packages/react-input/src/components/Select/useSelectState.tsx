import * as React from 'react';
import type { SelectState } from './Select.types';
import { useControllableState, useMergedRefs, useId } from '@fluentui/react-utilities';
import type { MenuProps } from '@pongo-ui/react-menu';
import { MenuItemRadio } from '@pongo-ui/react-menu';
import { Chevron } from './Chevron';

export const useSelectState = (state: SelectState) => {
  const {
    defaultValue: originalDefaultValue,
    value: originalValue,
    appearance,
    danger,
    disabled,
    size,
    label,
    helperText,
    contentBefore,
    contentAfter,
    onCheckedValueChange,
  } = state;
  const { id } = state.select;

  const defaultValue = originalDefaultValue && { select: originalDefaultValue };
  const value = originalValue && { select: originalValue };

  const [currentValue, setCurrentValue] = useControllableState<{ select: string[] }>({
    defaultState: defaultValue,
    state: value,
    initialState: { select: ['segoe'] },
  });

  const [open, setOpen] = React.useState(false);

  const labelId = label ? useId('select-label', id) : undefined;
  const triggerRef = React.useRef<HTMLSelectElement>(null);

  const onChange: MenuProps['onCheckedValueChange'] = (ev, { checkedItems }) => {
    onCheckedValueChange?.(ev, { checkedItems: checkedItems });
    setCurrentValue({ select: checkedItems });
  };

  const onOpenChange: MenuProps['onOpenChange'] = (e, data) => {
    triggerRef?.current?.focus();
    setOpen(data.open);
  };

  const onSelectChange = () => {
    /** no op */
  };

  const onSelectMouseDown = (ev: React.MouseEvent<HTMLSelectElement, MouseEvent>) => {
    ev.preventDefault();
  };

  const onSelectKeyDown = (ev: React.KeyboardEvent<HTMLSelectElement>) => {
    // Prevent select tag from opening using keyboard commands
    switch (ev.key) {
      case 'ArrowUp': {
        ev.preventDefault();
      }
      case 'ArrowDown': {
        ev.preventDefault();
      }
    }
  };

  state.select.ref = useMergedRefs(state.select.ref, triggerRef);
  state.root.appearance = appearance;
  state.root.danger = danger;
  state.root.disabled = disabled;
  state.root.size = size;
  state.root.contentBefore = contentBefore;
  state.root.contentAfter = contentAfter;
  state.root.currentValue = currentValue.select[0];

  state.menu.open = open;
  state.menu.onOpenChange = onOpenChange;

  state.menuList.onCheckedValueChange = onChange;
  state.menuList.checkedValues = currentValue;

  state.select.value = currentValue.select[0];
  state.select.onChange = onSelectChange;
  state.select.onMouseDown = onSelectMouseDown;
  state.select.onKeyDown = onSelectKeyDown;

  if (label) {
    state.root.label = label;
    state.root.labelId = labelId;
    state.select.id = labelId;
  }

  if (helperText) {
    state.root.helperText = helperText;
    // state.root.helperTextId = helperTextId;
    // state.input['aria-describedby'] = helperTextId;
  }

  state.root.contentBefore = contentBefore;

  // Default to the Chevron unless otherwise specified
  if (!contentAfter) {
    state.root.contentAfter = <Chevron visible={open} />;
  } else {
    state.root.contentAfter = contentAfter;
  }

  /**
   * Generates native html options for the select component based on the child Options provided.
   */
  const generateOptions = React.useMemo(() => {
    const options = [<option value="" disabled key="default"></option>];
    const children: any = state.root.children;
    const length: number = children?.length;

    if (length) {
      for (let i = 0; i < length; i++) {
        if (typeof children[i] === typeof MenuItemRadio) {
          options.push(
            <option value={children[i].props.value} disabled key={i}>
              {children[i].props.children}
            </option>,
          );
        }
      }
    }

    return options;
  }, [state.root.children]);

  state.select.children = generateOptions;

  return state;
};
