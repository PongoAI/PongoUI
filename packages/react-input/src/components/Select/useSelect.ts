import * as React from 'react';
import { resolveShorthand, getNativeElementProps } from '@fluentui/react-utilities';
import { useSelectState } from './useSelectState';
import type { SelectProps, SelectState } from './Select.types';
import { InputWrapper } from '../InputWrapper';
import { Menu, MenuPopover, MenuTrigger, MenuList } from '@pongo-ui/react-menu';
import { useFocusableGroup } from '@fluentui/react-tabster';

export const useSelect = (props: SelectProps, ref: React.Ref<HTMLInputElement>): SelectState => {
  const {
    value,
    defaultValue,
    onCheckedValueChange,
    appearance = 'outline',
    contentBefore,
    contentAfter,
    size = 'medium',
    disabled,
    danger,
    label,
    placeholder,
    menu,
    menuTrigger,
    menuPopover,
    menuList,
    helperText,
    select,
  } = props;

  const groupFocusAttributes = useFocusableGroup({ tabBehavior: 'limitedTrapFocus' });

  const state: SelectState = {
    value,
    defaultValue,
    onCheckedValueChange,
    size,
    appearance,
    disabled,
    danger,
    contentBefore,
    contentAfter,
    helperText,
    label,
    placeholder,
    components: {
      root: InputWrapper,
      select: 'select',
      menu: Menu,
      menuTrigger: MenuTrigger,
      menuPopover: MenuPopover,
      menuList: MenuList,
    },
    root: getNativeElementProps('div', {
      ref,
      ...groupFocusAttributes,
      ...props,
    }),
    select: resolveShorthand(select, { required: true }),
    menu: resolveShorthand(menu, { required: true }),
    menuTrigger: resolveShorthand(menuTrigger, { required: true }),
    menuPopover: resolveShorthand(menuPopover, { required: true }),
    menuList: resolveShorthand(menuList, { required: true }),
  };

  useSelectState(state);

  return state;
};
