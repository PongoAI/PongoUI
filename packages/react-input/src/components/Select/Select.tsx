import * as React from 'react';
import { useSelect } from './useSelect';
import { renderSelect } from './renderSelect';
import { useSelectStyles } from './useSelectStyles';
import type { SelectProps } from './Select.types';
import type { ForwardRefComponent } from '@fluentui/react-utilities';

/**
 * The Select component allows for users to choose an option from a dropdown menu.
 */
export const Select: ForwardRefComponent<SelectProps> = React.forwardRef((props, ref) => {
  const state = useSelect(props, ref);

  useSelectStyles(state);

  return renderSelect(state);
});
Select.displayName = 'Select';
