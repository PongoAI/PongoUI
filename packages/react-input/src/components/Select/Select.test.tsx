import React from 'react';
import { Select } from './Select';
import { Option } from '../Option';
import { SelectProps } from './Select.types';
import { isConformant } from '../../common/isConformant';
import { render } from '@testing-library/react';

describe('Select', () => {
  isConformant({
    Component: Select as React.FunctionComponent<SelectProps>,
    displayName: 'Select',
    disabledTests: ['make-styles-overrides-win'],
  });

  it('renders a basic Select', () => {
    const renderedComponent = render(<Select />);

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });

  it('renders a basic Select with an option', () => {
    const renderedComponent = render(
      <Select>
        <Option value="test">Test</Option>
      </Select>,
    );

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });

  it('renders a basic Select with an option and Header', () => {
    // This is checked to ensure it doesn't attempt to generate an item for anything other than an Option
    const renderedComponent = render(
      <Select>
        <h1>This isa test</h1>
        <Option value="test">Test</Option>
      </Select>,
    );

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });

  it('renders a Select contentBefore and contentAfter', () => {
    const renderedComponent = render(
      <Select contentBefore="hello" contentAfter="test">
        <h1>This isa test</h1>
        <Option value="test">Test</Option>
      </Select>,
    );

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });

  it('renders a disabled Select and applies prop to native select element', () => {
    const renderedComponent = render(
      <Select contentBefore="hello" contentAfter="test" disabled>
        <h1>This isa test</h1>
        <Option value="test">Test</Option>
      </Select>,
    );

    expect(renderedComponent.baseElement).toMatchSnapshot();
  });
});
