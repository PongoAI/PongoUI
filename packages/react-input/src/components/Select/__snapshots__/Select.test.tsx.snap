// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders a Select contentBefore and contentAfter 1`] = `
<body>
  <div>
    <span
      aria-haspopup="menu"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      id="menu10"
    >
      <span
        class="pongo-input-wrapper ___vwtn0r0 f10pi13n f1ewtqcl f22iagw f122n59 flvyvdh fc467gg f8qjfyk f8dqhol f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw fvfw249 f45u8dm f1arxnym f151knc1 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo fmw406p f12c27rx f17thq47 f12kdog7 f5qrn4q f1kndqco f1kc1xys f1d7l9b4 f16n8puj f1ci6z1z flfiaez f170vdtw fg4eyww f1cngt4b f17x5ctf fbk6y72 f1uajmgh f11t0hz4 f2v3se5 f18saz9o f80sxui f19q3xa0 f1vx0exu f1nae43o"
      >
        <span
          class="___1rw3gfb f1ewtqcl fyuhxbn f1ysjbbf f1qo5dib frdkuqy"
        >
          hello
        </span>
        <select
          class="___1irr20y f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx fy77jfu f1l02sjl f3rmtva f2hkw1w fakqst0 fdm5f4k f1fr6u07 f1k6fduh fc467gg f1vw2ftv f13ta7ih f1arxnym"
        >
          <option
            disabled=""
            value=""
          />
          <option
            disabled=""
          >
            This isa test
          </option>
          <option
            disabled=""
            value="test"
          >
            Test
          </option>
        </select>
        <span
          class="___12yahoq f1ewtqcl fyuhxbn f1ysjbbf f1qo5dib f81rol6"
        >
          test
        </span>
      </span>
    </span>
  </div>
</body>
`;

exports[`Select renders a basic Select 1`] = `
<body>
  <div>
    <span
      aria-haspopup="menu"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      id="menu7"
    >
      <span
        class="pongo-input-wrapper ___vwtn0r0 f10pi13n f1ewtqcl f22iagw f122n59 flvyvdh fc467gg f8qjfyk f8dqhol f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw fvfw249 f45u8dm f1arxnym f151knc1 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo fmw406p f12c27rx f17thq47 f12kdog7 f5qrn4q f1kndqco f1kc1xys f1d7l9b4 f16n8puj f1ci6z1z flfiaez f170vdtw fg4eyww f1cngt4b f17x5ctf fbk6y72 f1uajmgh f11t0hz4 f2v3se5 f18saz9o f80sxui f19q3xa0 f1vx0exu f1nae43o"
      >
        <select
          class="___1irr20y f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx fy77jfu f1l02sjl f3rmtva f2hkw1w fakqst0 fdm5f4k f1fr6u07 f1k6fduh fc467gg f1vw2ftv f13ta7ih f1arxnym"
        >
          <option
            disabled=""
            value=""
          />
        </select>
        <span
          class="___12yahoq f1ewtqcl fyuhxbn f1ysjbbf f1qo5dib f81rol6"
        >
          <svg
            class="___16pu995 f14t3ns0 fmrv4ls fuaaxji fq4mcun frvgh55"
            focusable="false"
            style="--pongo-icon-fill: var(--inherit);"
            viewBox="0 0 2048 2048"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              style="transform: translateY(10%); transition: transform .1s linear;"
            >
              <rect
                height="160"
                rx="80"
                style="transform-origin: center; transform: rotate(-45deg); transition: transform .1s linear;"
                width="908"
                x="941"
                y="944"
              />
              <rect
                height="160"
                rx="80"
                style="transform-origin: center; transform: rotate(45deg); transition: transform .1s linear;"
                width="908"
                x="198"
                y="944"
              />
            </g>
          </svg>
        </span>
      </span>
    </span>
  </div>
</body>
`;

exports[`Select renders a basic Select with an option 1`] = `
<body>
  <div>
    <span
      aria-haspopup="menu"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      id="menu8"
    >
      <span
        class="pongo-input-wrapper ___vwtn0r0 f10pi13n f1ewtqcl f22iagw f122n59 flvyvdh fc467gg f8qjfyk f8dqhol f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw fvfw249 f45u8dm f1arxnym f151knc1 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo fmw406p f12c27rx f17thq47 f12kdog7 f5qrn4q f1kndqco f1kc1xys f1d7l9b4 f16n8puj f1ci6z1z flfiaez f170vdtw fg4eyww f1cngt4b f17x5ctf fbk6y72 f1uajmgh f11t0hz4 f2v3se5 f18saz9o f80sxui f19q3xa0 f1vx0exu f1nae43o"
      >
        <select
          class="___1irr20y f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx fy77jfu f1l02sjl f3rmtva f2hkw1w fakqst0 fdm5f4k f1fr6u07 f1k6fduh fc467gg f1vw2ftv f13ta7ih f1arxnym"
        >
          <option
            disabled=""
            value=""
          />
        </select>
        <span
          class="___12yahoq f1ewtqcl fyuhxbn f1ysjbbf f1qo5dib f81rol6"
        >
          <svg
            class="___16pu995 f14t3ns0 fmrv4ls fuaaxji fq4mcun frvgh55"
            focusable="false"
            style="--pongo-icon-fill: var(--inherit);"
            viewBox="0 0 2048 2048"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              style="transform: translateY(10%); transition: transform .1s linear;"
            >
              <rect
                height="160"
                rx="80"
                style="transform-origin: center; transform: rotate(-45deg); transition: transform .1s linear;"
                width="908"
                x="941"
                y="944"
              />
              <rect
                height="160"
                rx="80"
                style="transform-origin: center; transform: rotate(45deg); transition: transform .1s linear;"
                width="908"
                x="198"
                y="944"
              />
            </g>
          </svg>
        </span>
      </span>
    </span>
  </div>
</body>
`;

exports[`Select renders a basic Select with an option and Header 1`] = `
<body>
  <div>
    <span
      aria-haspopup="menu"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      id="menu9"
    >
      <span
        class="pongo-input-wrapper ___vwtn0r0 f10pi13n f1ewtqcl f22iagw f122n59 flvyvdh fc467gg f8qjfyk f8dqhol f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw fvfw249 f45u8dm f1arxnym f151knc1 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo fmw406p f12c27rx f17thq47 f12kdog7 f5qrn4q f1kndqco f1kc1xys f1d7l9b4 f16n8puj f1ci6z1z flfiaez f170vdtw fg4eyww f1cngt4b f17x5ctf fbk6y72 f1uajmgh f11t0hz4 f2v3se5 f18saz9o f80sxui f19q3xa0 f1vx0exu f1nae43o"
      >
        <select
          class="___1irr20y f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx fy77jfu f1l02sjl f3rmtva f2hkw1w fakqst0 fdm5f4k f1fr6u07 f1k6fduh fc467gg f1vw2ftv f13ta7ih f1arxnym"
        >
          <option
            disabled=""
            value=""
          />
          <option
            disabled=""
          >
            This isa test
          </option>
          <option
            disabled=""
            value="test"
          >
            Test
          </option>
        </select>
        <span
          class="___12yahoq f1ewtqcl fyuhxbn f1ysjbbf f1qo5dib f81rol6"
        >
          <svg
            class="___16pu995 f14t3ns0 fmrv4ls fuaaxji fq4mcun frvgh55"
            focusable="false"
            style="--pongo-icon-fill: var(--inherit);"
            viewBox="0 0 2048 2048"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              style="transform: translateY(10%); transition: transform .1s linear;"
            >
              <rect
                height="160"
                rx="80"
                style="transform-origin: center; transform: rotate(-45deg); transition: transform .1s linear;"
                width="908"
                x="941"
                y="944"
              />
              <rect
                height="160"
                rx="80"
                style="transform-origin: center; transform: rotate(45deg); transition: transform .1s linear;"
                width="908"
                x="198"
                y="944"
              />
            </g>
          </svg>
        </span>
      </span>
    </span>
  </div>
</body>
`;

exports[`Select renders a disabled Select and applies prop to native select element 1`] = `
<body>
  <div>
    <span
      aria-haspopup="menu"
      data-tabster="{\\"groupper\\":{\\"tabbability\\":2}}"
      id="menu11"
    >
      <span
        class="pongo-input-wrapper ___ngrfis0 f10pi13n f1ewtqcl f22iagw f122n59 flvyvdh fc467gg f8qjfyk f8dqhol f4d9j23 f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp fifp7yv f1ov4xf1 f1asdtw4 f1ern45e f1n71otn f1h8hb77 f1deefiw fvfw249 f45u8dm f1arxnym f151knc1 fuzzvh5 fxf9f1y frdto5n ff0gctb f3zr33r f8g0ba1 f5uakvk frcipv9 f199l62q f1pa2qi2 fyqop9u f1ooiv2t f1bwkb58 fiio4mc f1m5ya7j fc8z0qo flm3n1z f1pltsua f15x2d8x fhngjcl f5qrn4q f1kndqco f1kc1xys f1d7l9b4"
      >
        <span
          class="___2yo2g90 f1ewtqcl fyuhxbn favx1wl f1hc2z27 frdkuqy"
        >
          hello
        </span>
        <select
          class="___18bnsog f1lmfglv fibjyge f1abmfm4 f9yszdx f1nbblvp f81rol6 f1ov4xf1 frdkuqy f1ern45e f1n71otn f1h8hb77 f1deefiw f1rbvai9 f13ju9t4 f1lh17z1 f1dlw1sp f1ewtqcl fqerorx fy77jfu f1l02sjl f3rmtva f2hkw1w fakqst0 fdm5f4k f1fr6u07 fdrzuqr favx1wl fgbidd3 f13ta7ih f1arxnym"
        >
          <option
            disabled=""
            value=""
          />
          <option
            disabled=""
          >
            This isa test
          </option>
          <option
            disabled=""
            value="test"
          >
            Test
          </option>
        </select>
        <span
          class="___popv1g0 f1ewtqcl fyuhxbn favx1wl f1hc2z27 f81rol6"
        >
          test
        </span>
      </span>
    </span>
  </div>
</body>
`;
